### DISCUSSION TITLE
is "has and belongs to many" possible?

### DISCUSSION THREAD

**Discussion by mirozahorak** [flutterdata/flutter_data#207] [Q&A]
i do not see any info about that in docs. 
is "has and belongs to many" possible?

**Comment by frank06** [comment]
Share an example please

**Reply by mirozahorak** [reply]
Thanks for the reply
basically the same functionality as in Ruby on Rails.
f.e. when i have a project and users, than to one project belong many users. and at the same time One user belongs to many projects. There is a third, connection table, connecting these 2 together via project_id and user_id.

you can fine more info about such here
https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association
and here
https://guides.rubyonrails.org/association_basics.html#the-has-and-belongs-to-many-association

**Reply by frank06** [reply]
Have you tried with a `HasMany` on each side?

**Reply by mirozahorak** [reply]
No, I have not, i cannot imagine how would that work. For "has many through" to work there is a need for separate connecting table and class with "belongs to" relations to separate classes. I am new to flutter_data and still trying to understand its inner workings.

**Reply by frank06** [reply]
Sure but this is not SQL. Best way is trying out and stepping through the code.

**Reply by mirozahorak** [reply]
I thought, because of ActiveRecord inspiration, this is somehow included and there is a certain level of "convention over configuration" in flutter_data already. 
But as i was thinking about this deeper, i see what kind of problems it can introduce. It is not a problem to implement such for a local adapters, but for server APIs it can be messy because they usually do not provide API access to the connecting table.
